File: templates/Uwp/_comp/_shared/Page.ImageGallery/Param_ProjectName.Core/Services/SampleDataService$wts.ItemName_gpostaction.cs
//{**
// This code block adds the method `GetImageGalleryDataAsync(string localResourcesPath)` to the SampleDataService of your project.
//**}
namespace Param_RootNamespace.Core.Services
{
    public static class SampleDataService
    {
//{[{
        private static ICollection<SampleImage> _gallerySampleData;
//}]}

        private static IEnumerable<SampleOrder> AllOrders()
        {
        }
//^^
//{[{

        // TODO WTS: Remove this once your image gallery page is displaying real data.
        public static async Task<IEnumerable<SampleImage>> GetImageGalleryDataAsync(string localResourcesPath)
        {
            if (_gallerySampleData == null)
            {
                _gallerySampleData = new List<SampleImage>();
                for (int i = 1; i <= 10; i++)
                {
                    _gallerySampleData.Add(new SampleImage()
                    {
                        ID = $"{i}",
                        Source = $"{localResourcesPath}/SampleData/SamplePhoto{i}.png",
                        Name = $"Image sample {i}"
                    });
                }
            }

            await Task.CompletedTask;
            return _gallerySampleData;
        }
//}]}
    }
}

File: templates/Uwp/_comp/_shared/Page.ImageGallery._VB/Param_ProjectName.Core/Services/SampleDataService$wts.ItemName_gpostaction.vb
'{**
' This code block adds the method `GetImageGalleryDataAsync(localResourcesPath As String)` to the SampleDataService of your project.
'**}
Namespace Services
    Public Module SampleDataService
'{[{
        Private _gallerySampleData As ICollection(Of SampleImage)
'}]}
        Public Function AllOrders() As IEnumerable(Of SampleOrder)
        End Function

        Public Function AllCompanies() As IEnumerable(Of SampleCompany)
        End Function
'^^
'{[{

        ' TODO WTS: Remove this once your image gallery page is displaying real data.
        Public Async Function GetImageGalleryDataAsync(localResourcesPath As String) As Task(Of IEnumerable(Of SampleImage))
            Await Task.CompletedTask
            If _gallerySampleData Is Nothing Then
                _gallerySampleData = New List(Of SampleImage)()
                For i As Integer = 1 To 10
                    _gallerySampleData.Add(New SampleImage() With {
                        .ID = $"{i}",
                        .Source = $"{localResourcesPath}/SampleData/SamplePhoto{i}.png",
                        .Name = $"Image sample {i}"
                    })
                Next
            End If

            Return _gallerySampleData
        End Function
'}]}
    End Module
End Namespace
